src/i18n/arabic.rs

```diff
@@ -0,0 +1,136 @@
+const NONE: u32 = u32::MAX;
+const AR_START: u32 = 0x621;
+const AR_END: u32 = 0x64A;
+const UNICODE_LAM: u32 = 0x644;
+
+pub fn fix_arabic_contextual_forms(chars: &mut [char]) {
```

Comment:
It appears the adjustment made to process the string array in-place generally functions well, except in scenarios where `is_lam_alif` returns true. To draw a parallel, consider "Lam" akin to the Latin "L" and "Alif" similar to the Latin "A." When "Alif" directly follows "Lam," they combine into a singular special Unicode character (akin to "LA"). Consequently, this combination results in the processed string potentially being shorter than its original. The revised method currently doesn't seem equipped to handle scenarios where the string length decreases. At the moment, I'm uncertain of the best approach to resolve this issue and don't have a specific solution to propose.

### How to reproduce
```
 cargo run --example banner 'لا'
```

### Expected:
```
       # 
       # 
   #   # 
    #  # 
     # # 
     ### 
     # # 
    #  ##
  ##     
```

### Actual
```
                
                
                
    #         # 
    #         # 
    #     #   # 
    #      #  # 
    #       # # 
    #       ### 
    #       # # 
    #      #  ##
         ##     
                
                
                
                

```